src/languages/yaml.js

```diff
@@ -10,6 +10,9 @@ Category: common, config
 export default function(hljs) {
   var LITERALS = 'true false yes no null';
 
+  // YAML spec allows non-reserved characters in tags
+  var YAML_TAG_RE = '[a-zA-Z_.~@#$%():;+=-?/]\\w*'
```

Comment:
So only the first character is the exception?  That doesn't seem right?

Reply:
Ah, you're right. This might work better
```suggestion
  var YAML_TAG_RE = '[a-zA-Z_.~@#$%():;+=-?/]+'
```

Reply:
Are there no rules about the first characters being different? Usually there is.

Reply:
Are spaces always required after `:` in YAML? Trying to wonder how `:` in a name isn't somewhat problematic... I guess because they are all escaped with `!`?

Reply:
Here's a link to the yaml spec for tags: https://yaml.org/spec/1.2/spec.html#id2782090
The intention is that they be able to encode URIs. Like you suspected, mappings do require a space after the colon.

Reply:
I'd stick with what we're talking about here unless you really want to go broader. The discussion is about local tags... did you mean to ALSO change data types?

```

ns-uri-char | ::= | “%” ns-hex-digit ns-hex-digit \| ns-word-char \| “#”\| “;” \| “/” \| “?” \| “:” \| “@” \| “&” \| “=” \| “+” \| “$” \| “,”\| “_” \| “.” \| “!” \| “~” \| “*” \| “'” \| “(” \| “)” \| “[” \| “]”
-- | -- | --



```

Reply:
In the spec, "data types" are just tags in a special namespace. The "secondary handle" !! is used to indicate this namespace, e.g. https://pyyaml.org/wiki/PyYAMLDocumentation#yaml-tags-and-python-types